docs/docs/building-modules/09-autocli.md

```diff
@@ -0,0 +1,129 @@
+---
+sidebar_position: 1
+---
+
+
+# AutoCLI
+
+:::note Synopsis
+This document details how to build CLI and REST interfaces for a module. Examples from various Cosmos SDK modules are included.
+:::
+
+:::note
+
+## Pre-requisite Readings
+
+* [Building Modules Intro](./01-intro.md)
+
+:::
+
+The `autocli` package is a [Go library](https://pkg.go.dev/cosmossdk.io/client/v2/autocli) for generating CLI (command line interface) interfaces for Cosmos SDK-based applications. It provides a simple way to add CLI commands to your application by generating them automatically based on your gRPC service definitions. Autocli generates CLI commands and flags directly from your protobuf messages, including options, input parameters, and output parameters. This means that you can easily add a CLI interface to your application without having to manually create and manage commands.
+
+## Getting Started
+
+Here are the steps to use the `autocli` package:
+
+1.  Define your app's modules that implement the `appmodule.AppModule` interface.
+2.  Create an instance of the `autocli.AppOptions` struct that specifies the modules you defined. 
+3. Use the EnhanceRootCommand() method provided by autocli to add the CLI commands for the specified modules to your root command.
+
+Here's an example of how to use `autocli`:
+
+``` go
+// Define your app's modules
+testModules := map[string]appmodule.AppModule{
+    "testModule": &TestModule{},
+}
+
+// Define the autocli AppOptions
+autoCliOpts := autocli.AppOptions{
+    Modules: testModules,
+}
+
+// Get the root command
+rootCmd := &cobra.Command{
+    Use: "app",
+}
+
+// Enhance the root command with autocli
+autocli.EnhanceRootCommand(rootCmd, autoCliOpts)
+
+// Run the root command
+if err := rootCmd.Execute(); err != nil {
+    fmt.Println(err)
+}
+```
+
+To use the autocli package, import it in your project and reference it in your code. For example, the `autocli.AppOptions` struct and `EnhanceRootCommand()` method can be found in the app.go file located in the `client/v2/autocli directory`.
```

Comment:
Can we precise `EnhanceRootCommand` does not create commands if it is already registered for a module, which means that it is additive only.